_source/_posts/2020-06-29-aspnet-core-mongodb.md

```diff
@@ -0,0 +1,513 @@
+
+The app you'll build is a virtual suggestion box. It is meant to strip away anything that doesn't deal with saving, retrieving, updating, and deleting data from the document store.
+
+For this example, you'll use MongoDB's [Atlas](https://www.mongodb.com/cloud/atlas). Atlas is a hosted solution for MongoDB clusters. You can easily sign up for a free account. You'll also use Okta for user authentication (you can also [sign up](https://developer.okta.com/signup) for a free account).
+
+## Scaffold Your ASP.NET Core Application
+
+For this application, I'll show you all the command-line commands for creating the application, coding in [VS Code](https://code.visualstudio.com/download). You could use Visual Studio but, if you're on macOS or Linux, you won't have that ability. 
+
+To scaffold a basic ASP.NET Core MVC application, use the following command:
+
+```sh
+dotnet new mvc --no-https -n OktaNetCoreMvcMongoExample
+```
+
+This command creates the application in a folder called `OktaNetCoreMvcMongoExample`. It also tells the .NET Core SDK not to use `https` for the development URL. Obviously, if this were a production application, you would want to leave that on. 
+
+Next, change into the `OktaNetCoreMvcMongoExample` folder and open VSCode in that directory. From there, install the two packages you need for the application to work:
+
+```sh
+dotnet add package MongoDB.Driver -v 2.10.4
+dotnet add package Microsoft.AspNetCore.Authentication.OpenIdConnect -v 3.1.4 
+```
+
+The `MongoDB.Driver` package allows .NET Core to connect to the MongoDB database. ou'll use the `Microsoft.AspNetCore.Authentication.OpenIdConnect` to authenticate to Okta.
+
+## Set Up Your Free MongoDB Cluster
+
+Once you've signed up for a free MongoDB cluster (I just used my Google account), you should be taken to a page that lists your clusters. The only cluster that should be in there to begin with is called **Cluster0**. Click on the **COLLECTIONS** button, then click the **+ Create Database** button and create a database called "SuggestMe" with a collection of "Submissions". You can have as many collections in the database as you want, but for the purposes of the "SuggestMe" app, all you'll need is the "Submissions" collection.
+
+{% img blog/aspnet-core-mongodb/mongo-create-db.png alt:"Create Mongo Database" width:"400" %}{: .center-image }
+
+## Connect Your .NET Core Application to the MongoDB Database
+
+Add your database connection information to the `appsettings.json` file in the root of your application. After the `Logging` section of the JSON document, add:
+
+```json
+,
```

Comment:
This leading comma looks funny. I'm guessing you're trying to tell developers they need it on the previous block. Personally, I'd remove it and expect developers to know. However, I can see why you'd add it too.